src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Collections
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.Collections, Categories.GenericCollections)]
+    [GenericTypeArguments(typeof(int), typeof(int))] // value type
+    [GenericTypeArguments(typeof(string), typeof(string))] // reference type
+    public class TryGetValueTrue<TKey, TValue>
+    {
+        private TKey[] _found;
+        private Dictionary<TKey, TValue> _source;
+        
+        private Dictionary<TKey, TValue> _dictionary;
+        private SortedList<TKey, TValue> _sortedList;
+        private SortedDictionary<TKey, TValue> _sortedDictionary;
+        private ConcurrentDictionary<TKey, TValue> _concurrentDictionary;
+        private ImmutableDictionary<TKey, TValue> _immutableDictionary;
+        private ImmutableSortedDictionary<TKey, TValue> _immutableSortedDictionary;
+
+        [Params(Utils.DefaultCollectionSize)]
+        public int Size;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _found = ValuesGenerator.ArrayOfUniqueValues<TKey>(Size);
+            _source = _found.ToDictionary(item => item, item => (TValue)(object)item);
+            _dictionary = new Dictionary<TKey, TValue>(_source);
+            _sortedList = new SortedList<TKey, TValue>(_source);
+            _sortedDictionary = new SortedDictionary<TKey, TValue>(_source);
+            _concurrentDictionary = new ConcurrentDictionary<TKey, TValue>(_source);
+            _immutableDictionary = Immutable.ImmutableDictionary.CreateRange<TKey, TValue>(_source);
+            _immutableSortedDictionary = Immutable.ImmutableSortedDictionary.CreateRange<TKey, TValue>(_source);
+        }
+
+        [Benchmark]
+        public bool Dictionary()
+        {
+            bool result = default;
+            var collection = _dictionary;
+            var found = _found;
```

Comment:
Nit: in most of this usage, var => whatever type it actually is

Reply:
do you mean that we should use the exact types instead?

Reply:
I have changed it to use the explicit types

Reply:
> do you mean that we should use the exact types instead

Yup. Thanks.